radixdlt-core/radixdlt/src/main/java/com/radixdlt/middleware2/network/GetVerticesRequestMessage.java

```diff
@@ -85,13 +87,15 @@
 	@DsonOutput(Output.ALL)
 	private final int count;
 
-	GetVerticesRequestMessage() {
-		// Serializer only
-		this.vertexId = null;
-		this.count = 0;
-	}
+	@JsonCreator
+	public GetVerticesRequestMessage(
+		@JsonProperty(value = "vertexId", required = true) HashCode vertexId,
+		@JsonProperty("count") int count
+	) {
+		if (count <= 0) {
+			throw new IllegalStateException("Request contains negative count of vertices: " + count);
```

Comment:
Also would rather see this as an exception like `IllegalArgumentException` rather than `IllegalStateException` as it isn't an invalid state to get to as a node can receive malformed requests.

Reply:
done